### ISSUE TITLE
FP 3925: Classes not having extra properties should not have to extend ISerializable interface

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#3945]
### Description
Rule 3925 demands that classes should always override `ISerializable` methods. However, if the class has no extra properties (or fields) this is not necessary, like in this scenario:

``` C#
public CustomLookup : Dictionary<string, object> // Compliant, no extra fields/properties to serialize
{
}
```

The rule should not be applied in cases like these.

**Comment by martin-strecker-sonarsource** [comment]
According to the [RSpec](https://sonarsource.github.io/rspec/#/rspec/S3925/csharp), this is indeed a false positive:

> * A derived type has serializable fields but the ISerializable.GetObjectData method is not overridden.

**Comment by Corniel** [comment]
I think this solution should also work.
